silo/src/cli.re

```diff
@@ -0,0 +1,79 @@
+open Printf;
+open Lexing;
+open Filename;
+open Arg;
+
+let version = () => {
+  let version =
+    switch (Build_info.V1.version()) {
+    | None => "unknown"
+    | Some(v) => Build_info.V1.Version.to_string(v)
+    };
+  print_endline(version);
+  exit(0);
+};
+
+let run = () => {
+  let name = ref(None);
+  let name_opt =
+    Cmd.opt(
+      ~names=["--name"],
+      ~doc="Set the package name, defaults to the directory name",
+      ~usage="<path>",
+      ~value=name,
+      ~conv=name =>
+      Some(name)
+    );
+  let path = ref("");
+  let path_arg =
+    Cmd.arg(
+      ~name="path", ~doc="Path for the new project", ~value=path, ~conv=path =>
+      path
+    );
+  let new_ =
+    Cmd.make(
+      ~name="new",
+      ~doc="Create a new Grain project",
+      ~usage="[options] <path>",
+      ~options=[name_opt],
+      ~args=[path_arg],
+      Cmd.Thunk(() => New.new_(~name=?name^, path^)),
+    );
+
+  let profile = ref(Build.Dev);
+  let release_opt =
+    Cmd.flag(
+      ~names=["--release"],
```

Comment:
Why is this a CLI flag rather than something in the config TOML?

Reply:
I assume this is a cli flag over being in the config because the average workflow would be compiling normally and then `release` is only used when your building for production at the end. 

What if instead of having this as a cli flag though we had build profiles in the config. so you could `silo build dev` or `silo build release`, (We could autogenerate a dev and release build, and have the regular build be the development one)